### ISSUE TITLE
reducedim does not work without an initial parameter for the max function

### ISSUE THREAD

**Issue by aoboturov** [JuliaLang/julia#16133]
`reducedim` does not work without an initial parameter for the `max` function (maybe also for some others). Although [documentation](http://docs.julialang.org/en/release-0.4/stdlib/arrays/#Base.reducedim) claims that it does.

Example to reproduce:

    using Distributions
    rn = Dirichlet(5, .5)
    xx = rand(rn, 1000)
    reducedim(max, xx, [1 2])

Produces an error:

    ERROR: MethodError: `reducedim_init` has no method matching reducedim_init(::Base.IdFun,   ::Function, ::Array{Float64,2}, ::Array{Int64,2})
    Closest candidates are:
    reducedim_init(::Base.IdFun, ::Base.AddFun, ::Union{AbstractArray{Complex{Float32},N},AbstractArray{Complex{Float64},N},AbstractArray{Complex{Int128},N},AbstractArray{Complex{Int64},N},AbstractArray{Complex{UInt128},N},AbstractArray{Complex{UInt64},N},AbstractArray{Float32,N},AbstractArray{Float64,N},AbstractArray{Int128,N},AbstractArray{Int64,N},AbstractArray{UInt128,N},AbstractArray{UInt64,N}}, ::Any)
    reducedim_init{S}(::Any, ::Base.AddFun, ::AbstractArray{S,N}, ::Any)
    reducedim_init(::Base.IdFun, ::Base.MulFun, ::Union{AbstractArray{Complex{Float32},N},AbstractArray{Complex{Float64},N},AbstractArray{Complex{Int128},N},AbstractArray{Complex{Int64},N},AbstractArray{Complex{UInt128},N},AbstractArray{Complex{UInt64},N},AbstractArray{Float32,N},AbstractArray{Float64,N},AbstractArray{Int128,N},AbstractArray{Int64,N},AbstractArray{UInt128,N},AbstractArray{UInt64,N}}, ::Any)
    ...
    in reducedim at reducedim.jl:247

If an initial value is provided reduction works

    reducedim(max, xx, [1 2], -1.)
    1x1 Array{Float64,2}:
     0.998208

**Comment by timholy** [comment]
Fixed in #16136. Thanks for the report, @aoboturov!

**Comment by colbec** [comment]
@timholy In openSUSE I'm getting this during `make`, but `make testall` runs correctly:

```
reduce.jl
WARNING: New definition 
    r_promote(Core.Inference.#max, Any) at reduce.jl:31
is ambiguous with: 
    r_promote(Any, Union{UInt32, Float16, UInt16, Int32, Int16, UInt8, Int8}) at reduce.jl:19.
To fix, define 
    r_promote(Core.Inference.#max, Union{UInt32, Float16, UInt16, Int32, Int16, UInt8, Int8})
before the new definition.
WARNING: New definition 
    r_promote(Core.Inference.#min, Any) at reduce.jl:32
is ambiguous with: 
    r_promote(Any, Union{UInt32, Float16, UInt16, Int32, Int16, UInt8, Int8}) at reduce.jl:19.
To fix, define 
    r_promote(Core.Inference.#min, Union{UInt32, Float16, UInt16, Int32, Int16, UInt8, Int8})
before the new definition.
intset.jl

```

**Comment by timholy** [comment]
Rats. Fixed in 46add3bfe9e9760c9159e6938d8b3c1a62b25076.
